ext/pdo_pgsql/pgsql_driver.c

```diff
@@ -38,8 +38,14 @@ static bool pgsql_handle_in_transaction(pdo_dbh_t *dbh);
 
 static char * _pdo_pgsql_trim_message(const char *message, int persistent)
 {
-	size_t i = strlen(message)-1;
+	size_t i = strlen(message);
 	char *tmp;
+	if (i == 0) {
```

Comment:
nit:

```c
if (UNEXPECTED(i == 0)) {
```

Reply:
Changed.

But in my opinion, things like this should be done with PGO, not by manual.